### ISSUE TITLE
HEVC HDR encoder does not add "Matrix coefficients" properly set in VUI

### ISSUE THREAD

**Issue by onealexanderivanov** [GPUOpen-LibrariesAndSDKs/AMF#269]
I have noticed that streams saved after AMF HEVC HDR encoding are NOT having "Matrix coefficients" set to value BT.2020 non-constant. Even if I set the following properties for output on HEVC Encoder:

```
enc->SetProperty(AMF_VIDEO_ENCODER_HEVC_NOMINAL_RANGE, AMF_VIDEO_ENCODER_HEVC_NOMINAL_RANGE_STUDIO);
enc->SetProperty(AMF_VIDEO_ENCODER_HEVC_COLOR_BIT_DEPTH, AMF_COLOR_BIT_DEPTH_10);
enc->SetProperty(AMF_VIDEO_ENCODER_HEVC_OUTPUT_COLOR_PROFILE, AMF_VIDEO_CONVERTER_COLOR_PROFILE_2020);
enc->SetProperty(AMF_VIDEO_ENCODER_HEVC_OUTPUT_COLOR_PRIMARIES, AMF_COLOR_PRIMARIES_BT2020);
enc->SetProperty(AMF_VIDEO_ENCODER_HEVC_OUTPUT_TRANSFER_CHARACTERISTIC, AMF_COLOR_TRANSFER_CHARACTERISTIC_SMPTE2084);
```

The problem is that YouTube does not recognize such stream as HDR during live streaming.
MediaInfo gives me the following characteristics for the same stream saved to file:

```
Video
ID : 1
Format : HEVC
Format/Info : High Efficiency Video Coding
Format profile : Main 10@L4.1@Main
HDR format : SMPTE ST 2086, HDR10 compatible
Codec ID : V_MPEGH/ISO/HEVC
Duration : 15 s 466 ms
Width : 1 920 pixels
Height : 1 080 pixels
Display aspect ratio : 16:9
Frame rate mode : Constant
Frame rate : 59.940 (60000/1001) FPS
Color space : YUV
Chroma subsampling : 4:2:0
Bit depth : 10 bits
Default : Yes
```

Could you suggest what I need to do to add into output stream's VUI the characteristic "Matrix coefficients" set to value BT.2020 non-constant

**Comment by rhutsAMD** [comment]
While the parameter in the VUI header is optional, we will start filling it in. This change will be included in an upcoming driver update.

No actions are needed from your side, and we will notify you once the driver with the fix is made publicly available.

**Comment by rhutsAMD** [comment]
The fix for the optional matrix coefficient in the VUI header is now publicly released in the Radeon™ Software Adrenalin 21.10.1 driver and I confirmed it is working with this driver.

**Comment by ByteYIFFF** [comment]
@rhutsAMD the video has hdr effect only for the first few seconds in rx6600m graphics card. this is my video.my amf version is 1.4.23 and driver is the latest version.
[hdr.zip](https://github.com/GPUOpen-LibrariesAndSDKs/AMF/files/8367904/hdr.zip)

In addition to depth, I also set color information parameters such as color_trc, color_pri, etc. But there are not color_trc, color_pri in the VUI steam by elecard hevc analzyer.
